include/zephyr/drivers/gpio.h

```diff
@@ -417,6 +417,247 @@ struct gpio_dt_spec {
 #define GPIO_DT_SPEC_INST_GET_OR(inst, prop, default_value) \
 	GPIO_DT_SPEC_INST_GET_BY_IDX_OR(inst, prop, 0, default_value)
 
+/*
+ * @cond INTERNAL_HIDDEN
+ *
+ * Auxiliary conditional macro that generates a bitmask for the range
+ * from @p "prop" array defined by the (off_idx, sz_idx) pair,
+ * or 0 if the range does not exist.
+ *
+ * @param node_id devicetree node identifier
+ * @param prop lowercase-and-underscores array property name
+ * @param off_idx logical index of bitmask offset value into "prop" array
+ * @param sz_idx logical index of bitmask size value into "prop" array
+ */
```

Comment:
Please separate the `@cond` statements from the rest of the comments as this allows for restructuring the functions/comments without accidentally moving the `@cond` to cover something else.

```suggestion
/**
 * @cond INTERNAL_HIDDEN
 *
 * For internal driver use only, skip these in public documentation.
 */

/** 
 * Auxiliary conditional macro that generates a bitmask for the range
 * from @p "prop" array defined by the (off_idx, sz_idx) pair,
 * or 0 if the range does not exist.
 *
 * @param node_id devicetree node identifier
 * @param prop lowercase-and-underscores array property name
 * @param off_idx logical index of bitmask offset value into "prop" array
 * @param sz_idx logical index of bitmask size value into "prop" array
 */
```

Reply:
done